sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/catalog/SessionCatalog.scala

```diff
@@ -693,7 +697,6 @@ class SessionCatalog(
       dropDatabase(db, ignoreIfNotExists = false, cascade = true)
     }
     tempTables.clear()
-    functionRegistry.clear()
```

Comment:
A long story to explain... : ) 
I need to add it back in the following fix: https://github.com/apache/spark/pull/12081/commits/a2d71fdb1af71d67b744b2f89a08d12837559c9b

You are right, I need to add it back. : )